src/lib/cached/cached.ml

```diff
@@ -170,24 +191,61 @@ let run
               None )
   with
   | Some x ->
-      return x
+      return (x, `Cache_hit)
   | None -> (
       Core.printf
         "Could not load %s from the following path:\n\
         \ \n\
          %s\n\
          %s\n\
+         %s\n\
         \ \n\
-        \ Trying the autogen path %s...\n"
-        name manual_install_path brew_install_path autogen_path ;
-      let base_path = base_path autogen_path in
-      match%bind With_components.load load ~base_path with
+        \ Trying to hit the s3 bucket...\n"
+        name manual_install_path brew_install_path s3_install_path ;
+      (* Attempt load from s3 *)
+      let open Cohttp_async in
+      let open Deferred.Let_syntax in
+      match%bind
+        let open Deferred.Or_error.Let_syntax in
+        let%bind () =
+          Deferred.Result.join
+          @@ Monitor.try_with (fun () ->
+                 let open Deferred.Let_syntax in
+                 let%bind resp, body =
+                   Client.get
+                     (Uri.of_string (base_path "todo/path/to/s3/bucket"))
```

Comment:
How do we know the object name?

eg,
https://s3-us-west-2.amazonaws.com/snark-keys.o1test.net/24700a6aa2cff39cbba18a2083c3c3f6_wrap_proving